File: cineManagement/cineManagementDatabaseFirst/Models/DTOs/PeliculaSalaCineDTO.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace cineManagementDatabaseFirst.Models.DTOs
{
    public class PeliculaSalaCineDTO
    {
        public int PeliculaSalaCineId { get; set; }
        public int PeliculaId { get; set; }
        public int SalaId { get; set; }
        public DateTime FechaPublicacion { get; set; }
        public DateTime FechaFin { get; set; }
        public PeliculaDTO Pelicula { get; set; }
        public SalaCineDTO Sala { get; set; }
    }

    public class PeliculaSalaCineCreateDTO
    {
        [Required]
        public int PeliculaId { get; set; }
        [Required]
        public int SalaId { get; set; }
        [Required]
        public DateTime FechaPublicacion { get; set; }
        [Required]
        public DateTime FechaFin { get; set; }
    }

    public class PeliculaSalaCineUpdateDTO
    {
        [Required]
        public int PeliculaSalaCineId { get; set; }

        [Required]
        public int PeliculaId { get; set; }

        [Required]
        public int SalaId { get; set; }

        [Required]
        public DateTime FechaPublicacion { get; set; }

        [Required]
        public DateTime FechaFin { get; set; }
    }
}

File: cineManagementFront/src/app/models/dtos/peliculaSalaCine.dto.ts
import { PeliculaDTO } from "./pelicula.dto";
import { SalaCineDTO } from "./salaCine.dto";

export interface PeliculaSalaCineDTO {
  peliculaSalaCineId: number;
  peliculaId: number;
  salaId: number;
  fechaPublicacion: Date;
  fechaFin: Date;
  pelicula?: PeliculaDTO;
  sala?: SalaCineDTO;
}

export interface PeliculaSalaCineCreateDTO {
  peliculaId: number;
  salaId: number;
  fechaPublicacion: Date;
  fechaFin: Date;
}

export interface PeliculaSalaCineUpdateDTO {
  peliculaSalaCineId: number;
  peliculaId: number;
  salaId: number;
  fechaPublicacion: Date;
  fechaFin: Date;
}
